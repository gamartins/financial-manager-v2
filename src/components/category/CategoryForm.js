import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

export default class CategoryForm extends React.Component {

  onSubmit(values, actions) {
    this.props.saveItem(values).then(() => actions.resetForm())
  }

  renderForm(props) {
    return (
      <form onSubmit={props.handleSubmit} className="form-inline">
        <div className='mr-3'>
          <input
            className="form-control"
            type="text"
            name="name"
            placeholder="Nome do item"
            onChange={props.handleChange}
            onBlur={props.handleBlur}
            value={props.values.name} />
        </div>
        
        <button className="btn btn-primary" type="submit">Salvar</button>
      </form>
    )
  }

  render() {
    return (
      <div className="mb-4">
        <Formik
          initialValues={this.props.initialValues}
          enableReinitialize={true}
          onSubmit={(values, actions) => this.onSubmit(values, actions)}>
            { props => this.renderForm(props) }
        </Formik>
      </div>
    )
  }
}

CategoryForm.propTypes = {
  initialValues: PropTypes.shape({
    name: PropTypes.string,
  }).isRequired,
}
