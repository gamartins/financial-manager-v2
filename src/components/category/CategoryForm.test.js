import React from 'react';
import { act } from 'react-dom/test-utils';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CategoryForm from './CategoryForm'

Enzyme.configure({ adapter: new Adapter() });

const initialValues = {
  id: null,
  name: '',
}

describe('<CategoryForm />', () => {
  test('should be created', () => {
    const wrapper = mount(<CategoryForm initialValues={initialValues} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should call saveItem prop on save button click', async () => {
    const saveItem = jest.fn(() => Promise.resolve())
    const component = mount(<CategoryForm
      initialValues={initialValues}
      saveItem={values => saveItem(values)}/>)

    await act(async () => {
      component.find('button').simulate('submit', { preventDefault: () => {} })
    })

    expect(saveItem).toHaveBeenCalledWith(initialValues)
  })

})