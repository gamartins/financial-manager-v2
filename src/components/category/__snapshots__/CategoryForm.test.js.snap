// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryForm /> should be created 1`] = `ReactWrapper {}`;
